src/cpp/session/projects/SessionProjectContext.cpp

```diff
@@ -54,6 +55,34 @@ namespace projects {
 
 namespace {
 
+static std::unique_ptr<r_util::RPackageInfo> s_pIndexedPackageInfo = nullptr;
+
+void onDescriptionChanged()
+{
+   s_pIndexedPackageInfo.reset();
+
+   std::unique_ptr<r_util::RPackageInfo> pInfo(new r_util::RPackageInfo);
+   Error error = pInfo->read(projectContext().directory());
+   if (error)
+   {
+      LOG_ERROR(error);
+      return;
+   }
+
+   pInfo.swap(s_pIndexedPackageInfo);
+}
+
+void onProjectFilesChanged(const std::vector<core::system::FileChangeEvent>& events)
+{
+   FilePath descPath = projectContext().directory().childPath("DESCRIPTION");
+   for (auto& event : events)
+   {
+      auto& info = event.fileInfo();
+      if (info.absolutePath() == descPath.absolutePath())
+         onDescriptionChanged();
```

Comment:
Nit: should we break early here? (presumably there can be only one event related to `DESCRIPTION`)

Reply:
Thanks for catching that! I agree we should break here.